src/librustc_typeck/diagnostics.rs

```diff
@@ -1258,18 +1262,53 @@ extern "rust-intrinsic" {
 "##,
 
 E0107: r##"
-This error means that an incorrect number of lifetime parameters were provided
-for a type (like a struct or enum) or trait:
+This error means that an incorrect number of type or lifetime parameters
```

Comment:
Since we're going to be getting const parameters fairly soon, it's easier just to call them "generic arguments" rather than "type or lifetime parameters".